source/fuzz/fuzzer_pass_add_access_chains.cpp

```diff
@@ -132,20 +137,21 @@ void FuzzerPassAddAccessChains::Apply() {
             break;
           }
 
-          // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3179) We
-          //  could allow non-constant indices when looking up non-structs,
-          //  using clamping to ensure they are in-bounds.
           uint32_t index_value =
               GetFuzzerContext()->GetRandomIndexForAccessChain(bound);
-          index_ids.push_back(FindOrCreateIntegerConstant(
-              {index_value}, 32, GetFuzzerContext()->ChooseEven()));
+
           switch (subobject_type->opcode()) {
             case SpvOpTypeArray:
             case SpvOpTypeMatrix:
             case SpvOpTypeVector:
+              // The index will be clamped
+              index_ids.push_back(FindOrCreateIntegerConstantReadyForClamping(
+                  index_value, bound, &fresh_ids_for_clamping));
```

Comment:
I actually think it's a bad idea to introduce the function FindOrCreateIntegerConstantReadyForClamping.

- The function is not very cohesive.  It adds a boolean type, adds an integer constant, returns the id of that constant, and then - on the side - creates a pair of fresh ids.

- The function is only called once.  Thus there's no argument for avoiding code duplication in the fuzzer pass.

- The function is very unlikely to be reusable, due to it being very specific to this fuzzer pass and not very cohesive overall.

- The body of the function would arguably be easier to understand if presented inline, here, rather than on its own.

Can you present the code inline and get rid of the function?

Reply:
Done